### PR TITLE
Localnode test bug in travis

### PR DESCRIPTION
**Pull request by y0sher**
Reopened pr to test the issue

### CODE REVIEW

**Comment by y0sher** [comment]
Something weird was happening. even though we created a node and all the file writing operations went well (no exceptions) after waiting 5 seconds (in the test) travis won't find this file. 
this never happens locally.

this maybe happened because travis ran tests simultaneously and file ops clashed. 
I added some security to make sure our files are written and read to/from filesystem.
though this didn't help and the same thing kept happening. (though pretty rare I have to run travis alot)

I've changed the data path for this test before deleting and creating files.
the localnodestore_test seems to pass every time now ( at least 10 times for now )
I think this might be a good practice for tests in the future anyway to not pollute the app main data path and make an isolated env for each test.

Bottom line I think this change can be approved. and we should progress to solve other race conditions that make travis fail. (mostly in p2p package).

**Comment by avive** [comment]
> I think this might be a good practice for tests in the future anyway to not pollute the app main data path and make an isolated env for each test.

I'm all for complete test env isolation for each test. Let's totally do it!
